packages/vscode-extension/src/KogitoEditor.ts

```
@@ -41,7 +41,13 @@ export class KogitoEditor implements EditorApi {
     private readonly messageBroadcaster: EnvelopeBusMessageBroadcaster,
     private readonly envelopeServer = new EnvelopeServer<KogitoEditorChannelApi, KogitoEditorEnvelopeApi>(
       {
-        postMessage: msg => this.panel.webview.postMessage(msg)
+        postMessage: msg => {
+          try {
+            this.panel.webview.postMessage(msg);
+          } catch (e) {
+            /* The webview has been disposed, thus cannot receive messages anymore. */
```

Is this really the only case in that this method can fail? If it isn't we will be masking error messages. Is there a way to not even call this method when after the webview is disposed? Looks like we're not "killing" everything properly before the Webview closes.

---

The only scenario that I noticed this fail was closing the editor while waiting for a promise. It's not specific to backend though. Afaik, it's just an annoying warning log on the console but the user wouldn't notice anything.
Maybe we can think of a promise cancelation mechanism.

I couldn't find a proper way to check if messages can be posted before posting it.
Please let me know if you have any suggestions.

---

Yes, good point. I'll create a JIRA for us to cancel whatever pending requests are registered on the EnvelopeMessageBusManager.